pkg/middlewares/aws_sns_topic_policy_expander_test.go

```diff
@@ -27,83 +27,198 @@ func TestAwsSNSTopicPolicyExpander_Execute(t *testing.T) {
 		{
 			name: "Inline policy no attached policy",
 			resourcesFromState: &[]resource.Resource{
-				&awsresource.AwsSnsTopic{
-					Arn:    aws.String("arn"),
-					Policy: aws.String("{\"policy\": \"coucou\"}"),
-					Id:     "ID",
+				&resource.AbstractResource{
+					Id:   "ID",
+					Type: awsresource.AwsSnsTopicResourceType,
+					Attrs: &resource.Attributes{
+						"arn":    "arn",
+						"id":     "ID",
+						"policy": "{\"policy\":\"coucou\"}",
+					},
 				},
 			},
 			expected: &[]resource.Resource{
-				&awsresource.AwsSnsTopic{
-					Arn:    aws.String("arn"),
-					Policy: nil,
-					Id:     "ID",
+				&resource.AbstractResource{
+					Id:   "ID",
+					Type: awsresource.AwsSnsTopicResourceType,
+					Attrs: &resource.Attributes{
+						"arn": "arn",
+						"id":  "ID",
+					},
 				},
-				&awsresource.AwsSnsTopicPolicy{
-					Arn:    aws.String("arn"),
-					Policy: aws.String("{\"policy\":\"coucou\"}"),
-					Id:     "ID",
+				&resource.AbstractResource{
+					Id:   "ID",
+					Type: awsresource.AwsSnsTopicPolicyResourceType,
+					Attrs: &resource.Attributes{
+						"arn":    "arn",
+						"id":     "ID",
+						"policy": "{\"policy\":\"coucou\"}",
+					},
 				},
 			},
 			mock: func(factory *terraform.MockResourceFactory) {
-				factory.On("CreateResource", mock.Anything, "aws_sns_topic_policy").Once().Return(nil, nil)
+				factory.On("CreateAbstractResource", "ID", awsresource.AwsSnsTopicPolicyResourceType, mock.MatchedBy(func(input map[string]interface{}) bool {
+					return input["id"] == "ID"
```

Comment:
Same remark as in another PR, we should change the ID's or either change the matcher

Reply:
I don't get this remark ? There is only one resource so only this id ?